From e45c1e2af5a75fffc6738b2a1c5e744d44aa47d6 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Thu, 12 Dec 2019 01:44:41 -0800
Subject: [PATCH] prevent TCP_NLA_* definition when netinet/tcp.h is included

---
 generic/include/linux/tcp.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/generic/include/linux/tcp.h b/generic/include/linux/tcp.h
index 0e4bd0c..f3224da 100644
--- a/generic/include/linux/tcp.h
+++ b/generic/include/linux/tcp.h
@@ -252,7 +252,16 @@ struct tcp_info {
 	__u32	tcpi_reord_seen;     /* reordering events seen */
 };
 
+#ifndef __UAPI_DEF_TCPHDR
+#ifdef _NETINET_TCP_H /* musl */
+#define __UAPI_DEF_TCP_NLA_PAD 0
+#else
+#define __UAPI_DEF_TCP_NLA_PAD 1
+#endif
+#endif
+
 /* netlink attributes types for SCM_TIMESTAMPING_OPT_STATS */
+#if __UAPI_DEF_TCP_NLA_PAD
 enum {
 	TCP_NLA_PAD,
 	TCP_NLA_BUSY,		/* Time (usec) busy sending data */
@@ -277,6 +286,7 @@ enum {
 	TCP_NLA_DSACK_DUPS,	/* DSACK blocks received */
 	TCP_NLA_REORD_SEEN,	/* reordering events seen */
 };
+#endif
 
 /* for TCP_MD5SIG socket option */
 #define TCP_MD5SIG_MAXKEYLEN	80
-- 
2.24.1

